pkg/workloads/console/controller.go

```diff
@@ -182,7 +182,14 @@ func (r *reconciler) Reconcile() (res reconcile.Result, err error) {
 		}
 	}
 
-	// Create or update the job
+	// Only create a job when the console is created. If the job is removed,
+	// return a reconciliation error.
+	if !r.console.Creating() {
+		if _, err = r.getJob(); err != nil {
+			return res, errors.Wrap(err, "failed to retrieve console job")
```

Comment:
It feels a bit odd to be returning a reconciliation error here, rather than continuing but logging a warning event, or updating the console's status (phase `Destroyed`?)

I think this may have a negative side-effect of firing alerts, because we have a metric for reconcile failures.

Reply:
Yeah I was 50/50 on this and wondering what people might think. 

My reason for the error is that deleting the job out from under the console is unexpected and shouldn't happen under normal operation. If we accept that statement as true, then I think we should error, fire alerts, be painful and attention seeking, when this happens. 

But I can see the flip side, where we say deleting a job is a valid way to kill a console, in which case we should mark the console as stopped (or more accurately like you mention, introduce a new phase called destroyed).

Having written this out now I think I have convinced myself of the latter and agree with you. We get to choose what is normal and supported behaviour and I can't see a downside to supporting the deletion of jobs.

I'll update this to introduce a destroyed phase which will be set if the job is missing with the same behaviour as the stopped phase - requeue after the TTL and and GC the console.